test/e2e/proxy.go

```diff
@@ -218,15 +230,24 @@ func proxyContext(version string) {
 			defer errLock.Unlock()
 			errs = append(errs, s)
 		}
-		for i := 0; i < proxyAttempts; i++ {
```

Comment:
I still would rather keep iterating over proxy attempts in the outer loop, then the varied requests in the inner loop.